src/Psalm/Config.php

```diff
@@ -955,6 +984,40 @@ private static function fromXmlAndPaths(
             $base_dir = $current_dir;
         }
 
+        $composer_json_path = Composer::getJsonFilePath($config->base_dir);
+
+        $composer_json = null;
+        if (file_exists($composer_json_path)) {
+            if (!$composer_json = json_decode(file_get_contents($composer_json_path), true)) {
+                throw new UnexpectedValueException('Invalid composer.json at ' . $composer_json_path);
+            }
+        }
+        foreach ($config->php_extensions as $ext => $_) {
+            $config->php_extensions[$ext] = isset($composer_json["require"]["ext-$ext"]);
```

Comment:
Est-ce qu'on peut ajouter require-dev et suggest ici? @VincentLanglet use-case is that he has a project where an extension is optional so it can't be added in require, the only place is in suggest

Reply:
I mentioned that earlier [here](https://github.com/vimeo/psalm/pull/7107#issuecomment-989871383). I think it'd be really nice to add better require-dev support in general, but that's outside the scope of this PR.
I think this case is likely uncommon enough that having `<enableExtensions>` and `<disableExtensions>` is enough to work around it. I'll think about it some more, maybe it wouldn't be a bad idea to include require-dev and suggests by default.

Reply:
I think we should allow what's in require-dev even outside tests. For example, if one has a logging extension, he may want to run it only on dev, but have code using it within optional checks. But you're right, config tag should be enough